File: 2089-maximum-matrix-sum/2089-maximum-matrix-sum.java
class Solution {
    public long maxMatrixSum(int[][] matrix) {
        long sum = 0;
        // for(int i = 0 ;i<matrix.length;i++){
        //     for(int j  = 0 ; j<matrix[0].length;j++){
        //         System.out.print(matrix[i][j]+" ");
        //     }
        //     System.out.println();
        // }
        int count = 0 ; 
        int min = Integer.MAX_VALUE ;
        for(int i  = 0 ; i<matrix.length ;i++){
            for(int  j = 0 ; j<matrix[0].length ;j++){
                sum +=Math.abs(matrix[i][j]);
                if(matrix[i][j]<0){
                    count++ ;
                }
                if(Math.abs(matrix[i][j])<min){
                    min = Math.abs(matrix[i][j]) ;
                }
            }
        }
        if(count%2!=0){
            sum -=2*min ;
        }
        return sum;
    }
}

File: 2089-maximum-matrix-sum/2089-maximum-matrix-sum.c
long long maxMatrixSum(int** matrix, int matrixSize, int* matrixColSize) {
     long long sum = 0;
    
        int count = 0 ; 
        int min = INT_MAX;
        int x =*matrixColSize ;
        printf("%d",x);
        for(int i  = 0 ; i<matrixSize ;i++){
            for(int  j = 0 ; j<x ;j++){
                sum +=abs(matrix[i][j]);
                if(matrix[i][j]<0){
                    count++ ;
                }
                if(abs(matrix[i][j])<min){
                    min = abs(matrix[i][j]) ;
                }
            }
        }
        if(count%2!=0){
            sum -=2*min ;
        }
        return sum;
}